Implement external CAD GLTF merging

1914Jegx opened the pull request:
## Summary
- add a GLTF loader and reuse glTF-Transform to merge external `cad_component` GLTF assets into the generated document
- extend the 3D converter to capture `model_gltf_url` entries, caching assets and preserving transforms
- add a poppygl-backed regression test + snapshot to ensure the merged models render correctly and face the expected direction

Fixes #14

## Testing
- bunx tsc --noEmit
- bun test --timeout 900000 tests/integration/multiple-cad-gltf.test.ts
- bun test --timeout 900000 tests/integration/circuit-to-gltf.test.ts
- bun test tests/board-texture-debug.test.ts

/claim #14

seveibar commented on lib/converters/circuit-to-3d.ts:
```diff
@@ -31,6 +30,99 @@ function convertRotationFromCadRotation(rot: {
+    ["cad_model", "gltf_url"],
+    ["cad_model", "gltfUrl"],
+    ["cadModel", "gltfUrl"],
+  ]
```
This is crazy. Gltf urls are literally ALWAYS cad_component.model_gltf_url or cad_component.model_glb_url

seveibar commented on lib/converters/scene-to-gltf.ts:
```diff
@@ -1,17 +1,317 @@
+  KHRDracoMeshCompression: unknown
+  meshoptDecoder?: MeshoptDecoderType
+  dracoDecoder?: unknown
+}
```
Why are you defining this? Remove, it is imported

seveibar commented on lib/converters/scene-to-gltf.ts:
```diff
@@ -1,17 +1,317 @@
+      const [core, functions, extensions] = await Promise.all([
+        import("@gltf-transform/core"),
+        import("@gltf-transform/functions"),
+        import("@gltf-transform/extensions"),
```
If you keep PRing dynamic importants we’re going to assume you arent reading the code

1914Jegx commented:
Thanks for the quick feedback, @seveibar! I’ve cleaned things up:

- `cad_component` GLTF detection now only looks at `model_gltf_url` / `model_glb_url` since that’s all we need.
- The scene exporter no longer uses dynamic imports. `@gltf-transform/*`, meshoptimizer, and draco are imported statically; we just configure the `NodeIO` instance once per run.
- Dropped the ad-hoc type definitions that were duplicating what those packages already export.

CI is green again (format/test/type-check). Let me know if there’s anything else you’d like me to adjust!

seveibar commented on lib/converters/circuit-to-3d.ts:
```diff
@@ -118,6 +159,32 @@ export async function convertCircuitJsonTo3D(
+      ? convertRotationFromCadRotation(cad.rotation)
+      : undefined
+
+    if (gltfUrl) {
```
Why is this being handled differently than other models

seveibar commented on lib/loaders/gltf.ts:
```diff
@@ -0,0 +1,184 @@
+}
+
+export async function loadGLTF(url: string): Promise<LoadedGLTFAsset> {
+  if (cache.has(url)) {
```
Theres a global cache here, but elsewhere you use a local cache…
